src/applicationManager.ts

```diff
@@ -528,3 +546,239 @@ async function getGraphAPIRoles(token: string, graphApiPermissions: string[]) {
+    ({ displayName, id }) => `name:${displayName}_id:${id}`,
+  );
+  const existingAppRolePairs = appRolesJson.map(
+    ({ displayName, id }) => `name:${displayName}_id:${id}`,
+  );
+  const commonPairs = existingAppRolePairs.filter((pair) =>
+    incomingAppRolePairs.includes(pair),
+  );
+
+  // Break here if app roles are the same as current file - not disabling all roles if unnecessary
+  if (!(commonPairs.length < existingAppRolePairs.length)) {
+    return;
+  }
+
+  // Keep fetched array of AppRoles but change enabled to false
+  const disabledAppRolesJson = appRolesJson.map(
+    (role) =>
+      ({
+        ...role,
+        isEnabled: false,
+      }) as AppRole,
+  );
+  await updateApplicationAppRoles({
+    token,
+    applicationId,
+    appRoles: disabledAppRolesJson,
+  });
+  console.log("Temporarily disabled app roles to allow updates");
+}
+
+/**
+ * Checks at least one of the group app role assignment mappings exists for a given entra group ID and app role ID, for a given application
+ */
+async function checkIfGroupAppRoleAssignmentExists({
+  token,
+  groupId,
+  appRoleId,
+  applicationId,
+}: {
+  token: string;
+  groupId: string;
+  appRoleId: string;
+  applicationId: string;
+}): Promise<boolean> {
+  const response = await fetch(
+    `https://graph.microsoft.com/v1.0/groups/${groupId}/appRoleAssignments`,
+    {
+      method: "GET",
+      headers: {
+        Authorization: `Bearer ${token}`,
+        "Content-Type": "application/json",
+      },
+    },
+  );
+  const assignments = await response.json();
+  const assignmentExists = assignments.value.some(
+    (assignment: { appRoleId: string; resourceId: string }) =>
+      assignment.appRoleId === appRoleId &&
+      assignment.resourceId === applicationId,
+  );
+  return assignmentExists;
+}
+
+/**
+ * Assigns groups to App Roles for a given application
+ * @param groupId Azure entra group ID to assign an app role to
+ * @param appRoleId ID of the app role to assign the group to
+ * @param applicationId ID of the application to assign the group to app roles for
+ */
+async function updateApplicationGroupAssignments({
+  token,
+  groupId,
+  appRoleId,
+  applicationId,
+}: {
+  token: string;
+  groupId: string;
+  appRoleId: string;
+  applicationId: string;
+}): Promise<void> {
+  const existingAssignment = await checkIfGroupAppRoleAssignmentExists({
+    token,
+    groupId,
+    appRoleId,
+    applicationId,
+  });
+
+  if (!existingAssignment) {
+    let body = {
+      principalId: groupId,
+      resourceId: applicationId,
+      appRoleId: appRoleId,
+    };
+    const result = await fetch(
+      `https://graph.microsoft.com/v1.0/groups/${groupId}/appRoleAssignments`,
+      {
+        method: "POST",
+        headers: {
+          Authorization: `Bearer ${token}`,
+          "Content-Type": "application/json",
+        },
+        body: JSON.stringify(body),
+      },
+    );
+    console.log("Added group role assignment");
```

Comment:
would be useful to log groupId, applicaitonId, and appRoleId

Reply:
There is already a good amount of info being logged when this runs - not sure printing out all the ideas would look nice in there - i.e from yarn test

```
Updated group role assignments for: Some name App Role, processing group: test_group_A
```